### ISSUE TITLE
problem with input focusing/opening dropdown

### ISSUE THREAD

**Issue by as090419** [angular-ui/ui-select#1560]
Problem occures when we click on input - it should be focused and i should be able to write something in input, but i cant - and what more - dropdown with suggestions becoming opened.

latest $animate is not working with ui-select, problem is here:

     if (ctrl.$animate && ctrl.$animate.on && ctrl.$animate.enabled(container[0])) {
//I CAN GO INHERE
        ctrl.$animate.on('enter', container[0], function (elem, phase) {
//BUT THIS BLOCK IS NOT REACHED
          if (phase === 'close') {
            // Only focus input after the animation has finished
            $timeout(function () {
              ctrl.focusSearchInput(initSearchValue);
            });
          }
        });

temporary fix that i made is commented block with if/else conditions replaced by:

        $timeout(function () {
          ctrl.focusSearchInput(initSearchValue);
          if(!ctrl.tagging.isActivated && ctrl.items.length > 1) {
            _ensureHighlightVisible();
          }
        });

Im working with angular 1.5.3 and latest version of ui-select.

**Comment by user378230** [comment]
This looks like a duplicate of #1306 and/or #1436 - can you confirm if that is the case?

**Comment by as090419** [comment]
No, its not. I can focus input but not on first click and dropdown shouldn't be opened without any available suggestions inside

**Comment by user378230** [comment]
Can you create a plunkr that demonstrates this? (Helpful starting point: http://angular-ui.github.io/ui-select/#examples)

**Comment by SergeyCherman** [comment]
Having the same issue, will report back if I can reproduce in a plunkr. In my case I have $animate.enabled(false) in my controller. If I set it to true then it works, however I need it set to false for other reasons.

**Comment by as090419** [comment]
I've created a new plunker that demonstates my issue (first dropdown, regular ui select):

http://plnkr.co/edit/X6eD7W2LwYG5j2yDdSme?p=preview

**Comment by hinok** [comment]
Hey,
The problem occurs also for selectize theme and ngAnimate when $animateProvider is configured.

```
app.config(function($animateProvider) {
  $animateProvider.classNameFilter(/my-animate/);
});
```

plnkr - http://plnkr.co/edit/mcFa2NyYsbULrTPsLr99?p=preview

**Comment by gatperdut** [comment]
having the same issue

**Comment by gasfab999** [comment]
Hey,
in our application we also have this problem.
Has someone found a workaround or is there already any progress in solving this problem with the next version?

**Comment by marcbachmann** [comment]
I'm currently monkey-patching the `$animate.enabled` method.
And I'm using `$animateProvider.classNameFilter` to enable animations on specific elements.

```js
var animationClassNameFilter = /vnd-animated/
angular.module('something').config(function ($animateProvider) {
    $animateProvider.classNameFilter(animationClassNameFilter)
})

angular.module('ui-select').run(function($animate) {
  var origEnabled = $animate.enabled
  $animate.enabled = function (elem) {
    if (arguments.length !== 1) {
      return origEnabled.apply($animate, arguments)
    } else if (origEnabled(elem)) {
      return animationClassNameFilter.test(elem.classNames)
    }
    return false
  }
})
```

```html
<div class="vnd-animated">This div uses animations</div>
```

**Comment by sgsvenkatesh** [comment]
I'm facing a similar issue if the dropdown has no content and append-to-body are used together. I think the dropdown is created and destroyed when there are no items in dropdown list, which is resulting focusing the body element.
**End result**: A click on input field, is focusing the body element for the first time. It works fine when the input field is clicked again.
